packages/flutter_tools/lib/src/commands/apk.dart

```diff
@@ -151,14 +180,41 @@ class ApkCommand extends FlutterCommand {
         help: 'Password for the entry within the keystore.');
   }
 
+  Future _findServices(_ApkComponents components) async {
+    if (!ArtifactStore.isPackageRootValid)
+      return;
+
+    dynamic manifest = _loadYamlFile(_kFlutterManifestPath);
+    if (manifest['services'] == null)
+      return;
+
+    for (String service in manifest['services']) {
+      File serviceRoot = '${ArtifactStore.packageRoot}/$service/apk';
+      dynamic serviceConfig = _loadYamlFile('$serviceRoot/config.yaml');
+      if (serviceConfig == null || serviceConfig['jars'] == null)
+        continue;
+      components.serviceClasses.add(serviceConfig['registration-class']);
+      for (String jar in serviceConfig['jars']) {
+        components.jars.add(new File('$serviceRoot/$jar'));
+      }
+    }
+  }
+
   Future<_ApkComponents> _findApkComponents(BuildConfiguration config) async {
     String androidSdkPath;
     List<String> artifactPaths;
     if (runner.enginePath != null) {
       androidSdkPath = '${runner.enginePath}/third_party/android_tools/sdk';
+      // TODO(mpcomplete): This might be messier than necessary. Questions:
+      // 1. Why does linking classes.dex + services.jar not work? I need to
+      // link classes.dex.jar + services.jar instead. (Maybe classes.dex is
+      // like an exe and classes.dex.jar is like a static lib?)
+      // 2. Is there a way to get sky/shell/BUILD.gn to generate
+      // classes.dex.jar as one file, instead of shell_lib.dex.jar + shell.jar?
```

Comment:
Probably we should have the engine repro produce a single jar and run dex on it here with all the other jars.